app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,57 @@
+class ReviewsController < ApplicationController
+  def new
+    @restaurant = Restaurant.find(params[:restaurant_id])
+    @review = Review.new
+  end
+
+  def create
+    @restaurant = Restaurant.find(params[:restaurant_id])
+    @review = @restaurant.reviews.new(review_params)
+    @review.user = current_user
+    if @review.save
+      flash[:notice] = "Review Created!"
+      redirect_to restaurant_path(@restaurant)
+    else
+      render :new
+    end
+  end
+
+  def destroy
+    @restaurant = Restaurant.find(params[:restaurant_id])
+    @review = Review.find(params[:id])
+
+    if @review.user_id == current_user.id
+      @review.destroy
+      flash[:notice] = "Review Deleted!"
+      redirect_to restaurant_path(@restaurant)
+    end
+  end
+
+  def edit
+    @restaurant = Restaurant.find(params[:restaurant_id])
+    @review = Review.find(params[:id])
+  end
+
+  def update
+    @review = Review.find(params[:id])
+    @restaurant = @review.restaurant
+
+    if @review.user_id != current_user.id
```

Comment:
consider an instance method for authorization here:

```ruby
@review.editable_by?(current_user)
```

You should also authorize before the edit view is rendered